src/dshould/thrown.d

```diff
@@ -0,0 +1,134 @@
+module dshould.thrown;
+
+import std.format : format;
+import std.traits : CommonType;
+import std.typecons;
+import dshould.ShouldType;
+
+unittest
+{
+    import dshould.basic : be, equal, not, should;
+
+    auto error = new Exception("");
+
+    /**
+     * Throws: Exception
+     */
+    void throwsException() { throw error; }
+
+    throwsException.should.throwAn!Exception.where.it.should.be(error);
+    throwsException.should.throwAn!Exception.where.it.should.not.be(null);
+
+    2.should.be(5).because("it just should, okay")
+        .should.throwA!FluentException.where.its.reason.should.equal("it just should, okay");
+
+    2.should.be(5).should.throwA!FluentException;
+    2.should.be(5).should.throwAn!Error.should.throwA!FluentException;
+
+    2.should.be(2).should.not.throwA!FluentException;
+
+    2.should.be(5).should.throwA!FluentException("test");
```

Comment:
?

Reply:
oops, that's leftovers. this only works in package.